### ISSUE TITLE
Add tests

### ISSUE THREAD

**Issue by Herschel** [Herschel/puremp3#1]
* Add test vectors using different formats. Verify against good output from minimp3.
* Add unit tests for specific algorithms.

**Comment by nico-abram** [comment]
Would fuzzing also be desireable?

**Comment by Herschel** [comment]
Yep, fuzzing would be great. Currently this crate is unmaintained (I'll add a note to the README), but I'll accept PRs if someone wanted to improve it.
